teps/00XX-interceptor-plugins.md

```diff
@@ -0,0 +1,423 @@
+of this file) is [here](/teps/NNNN-TEP-template/README.md).
+
+-->
+# TEP-NNNN: Pluggable Interceptors
+
+<!--
+This is the title of your TEP.  Keep it short, simple, and descriptive.  A good
+title can help communicate what the TEP is and should be considered as part of
+any review.
+-->
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a TEP and for
+highlighting any additional information provided beyond the standard TEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [TEP-NNNN: Pluggable Interceptors](#tep-nnnn-pluggable-interceptors)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Requirements](#requirements)
+  - [Proposal](#proposal)
+    - [Notes/Constraints/Caveats](#notesconstraintscaveats)
+    - [Risks and Mitigations](#risks-and-mitigations)
+    - [Performance (optional)](#performance-optional)
+  - [Design Details](#design-details)
+    - [HTTP Interface](#http-interface)
+      - [Request](#request)
+      - [Response](#response)
+    - [Installing Interceptors](#installing-interceptors)
+  - [Test Plan](#test-plan)
+  - [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+    - [Implement new HTTP interface without Interceptor CRD](#implement-new-http-interface-without-interceptor-crd)
+    - [Use go-plugin for built-in interceptors](#use-go-plugin-for-built-in-interceptors)
+    - [Allow operators to disable some built-in interceptors](#allow-operators-to-disable-some-built-in-interceptors)
+    - [Implement built-in interceptors using the current webhook interface](#implement-built-in-interceptors-using-the-current-webhook-interface)
+  - [Upgrade & Migration Strategy](#upgrade--migration-strategy)
+  - [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This is a Triggers proposal to make interceptors more pluggable and
+extensible by making the API uniform across built-in and webhook interceptors
+(i.e. using HTTP), and by updating the API to support some missing use cases.
+This should allow operators to select which interceptors they'd like to
+support as part of their installation. In addition, this allows for better
+auditability (see TEP-0022), the ability to distinguish between unexpected
+errors and filtered events (i.e. processing is successful but we don't need
+to continue; see [#336](https://github.com/tektoncd/triggers/issues/336)),
+and leaves the room open to easily extend the interceptor interface in the
+future.
+
+## Motivation
+Interceptors provide an extensible way to do event processing such as
+validating event payloads, filtering etc. When an EventListener receives an
+event, it will first process any configured interceptors for each trigger
+before further processing. Interceptors have two API surfaces:
```

Comment:
You have "Interceptor have two API surfaces:" repeated here and on the next line. Remove one?